tests/e2e/util.go

```diff
@@ -941,6 +941,44 @@ func updateCSIDeploymentTemplateFullSyncInterval(client clientset.Interface, min
 	return oldValue
 }
 
+// updateCSIDeploymentProvisionerTimeout helps to update the timeout value
+// in deployment template. For this to take effect,
+func updateCSIDeploymentProvisionerTimeout(client clientset.Interface, namespace string, timeout string) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ignoreLabels := make(map[string]string)
+	list_of_pods, err := fpod.GetPodsInNamespace(client, csiSystemNamespace, ignoreLabels)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	num_csi_pods := len(list_of_pods)
+
+	deployment, err := client.AppsV1().Deployments(namespace).Get(
+		ctx, vSphereCSIControllerPodNamePrefix, metav1.GetOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	containers := deployment.Spec.Template.Spec.Containers
+	for _, c := range containers {
+		if c.Name == "csi-provisioner" {
+			for i, e := range c.Args {
+				if strings.Contains(e, "timeout") {
+					if strings.Contains(e, timeout) {
+						framework.Logf("Timeout value is already set to expected value, hence return")
+						return
+					} else {
+						c.Args[i] = "--timeout=" + timeout + "s"
+					}
+				}
+			}
+		}
+	}
+	deployment.Spec.Template.Spec.Containers = containers
+	_, err = client.AppsV1().Deployments(namespace).Update(ctx, deployment, metav1.UpdateOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	framework.Logf("Waiting for a min for update operation on deployment to take effect...")
+	time.Sleep(1 * time.Minute)
```

Comment:
can we remove/reduce this?

Reply:
This wait is needed to make sure the update operation on the deployment takes effect. Reducing this to 30s might still have the previous pod status in running state.  1min is a safe wait time.